src/contractor/contractor.cpp

```diff
@@ -172,37 +185,74 @@ std::size_t Contractor::LoadEdgeExpandedGraph(
 
     std::unordered_map<std::pair<OSMNodeID, OSMNodeID>, std::pair<unsigned, uint8_t>>
         segment_speed_lookup;
+    std::unordered_map<std::tuple<OSMNodeID, OSMNodeID, OSMNodeID>, std::pair<double, uint8_t>>
+        turn_penalty_lookup;
 
     // If we update the edge weights, this file will hold the datasource information
     // for each segment
     std::vector<uint8_t> m_geometry_datasource;
 
-    if (update_edge_weights)
+    if (update_edge_weights || update_turn_penalties)
     {
-        uint8_t file_id = 1;
-        for (auto segment_speed_filename : segment_speed_filenames)
+        uint8_t segment_file_id = 1;
+        uint8_t turn_file_id = 1;
+
+        if (update_edge_weights)
         {
-            util::SimpleLogger().Write()
-                << "Segment speed data supplied, will update edge weights from "
-                << segment_speed_filename;
-            io::CSVReader<3> csv_in(segment_speed_filename);
-            csv_in.set_header("from_node", "to_node", "speed");
-            uint64_t from_node_id{};
-            uint64_t to_node_id{};
-            unsigned speed{};
-            while (csv_in.read_row(from_node_id, to_node_id, speed))
+            for (auto segment_speed_filename : segment_speed_filenames)
             {
-                segment_speed_lookup[std::make_pair(OSMNodeID(from_node_id),
-                                                    OSMNodeID(to_node_id))] =
-                    std::make_pair(speed, file_id);
+                util::SimpleLogger().Write()
+                    << "Segment speed data supplied, will update edge weights from "
+                    << segment_speed_filename;
+                io::CSVReader<3> csv_in(segment_speed_filename);
+                csv_in.set_header("from_node", "to_node", "speed");
+                uint64_t from_node_id{};
+                uint64_t to_node_id{};
+                unsigned speed{};
+                while (csv_in.read_row(from_node_id, to_node_id, speed))
+                {
+                    segment_speed_lookup[std::make_pair(OSMNodeID(from_node_id),
+                                                        OSMNodeID(to_node_id))] =
+                        std::make_pair(speed, segment_file_id);
+                }
+                ++segment_file_id;
+
+                // Check for overflow
+                if (segment_file_id == 0)
+                {
+                    throw util::exception(
+                        "Sorry, there's a limit of 254 segment speed files; you supplied too many");
```

Comment:
any reason to start at `1`, count until `0` and allow only `254`? Currently this should at least say 255?  (1...255 are valid values)

The same could be done with a proper `256` count, starting at 0.